sessions/test_js_challenge.py

```diff
@@ -325,3 +331,77 @@ def test_fail_challenge_too_late(self):
                                   delay_min=1000, delay_range=1000,
                                   status_code=503, expect_pass=False,
                                   req_delay=3, restart_on_fail=True)
+
+class JSChallengeAfterReload(BaseJSChallenge):
```

Comment:
The test checks that the JS challenge works after reload. Also need cases:
- JSCH was enabled (was listed), disabled on reconfig (not listed any more)
- JSCH was disabled, enabled on reconfig
- JSCH is compatible with redirect mark (last time I pached the sticky module I saw and fixed it)

Reply:
Hard to track the test cases, need to group them. And we both missed the case, which was originally reported by Alexander: JSCH was enabled and was enabled on reconfig (reload without config change).